db/migrate/20210419135355_create_categories.rb

```diff
@@ -0,0 +1,11 @@
+class CreateCategories < ActiveRecord::Migration[6.1]
+  def change
+    create_table :categories do |t|
+      t.string :title, null: false
```

Comment:
remove line#8 and add t.references :parent, foreign_key: { to_table: :categories }